apps/guide/src/content/03-creating-your-bot/04-handling-command-interactions.mdx

```diff
@@ -0,0 +1,292 @@
+---
+title: Handling command interactions
+category: Creating your bot
+---
+
+# Command handling
+
+Unless your bot project is small, it's not a very good idea to have a single file with a giant _`if`_/_`else if`_ chain for commands. If you want to implement features into your bot and make your development process a lot less painful, you'll want to implement a command handler. Let's get started on that!
+
+<Alert title="Read first!" type="info">
+	For fully functional slash commands, there are three important pieces of code that need to be written. They are:
+
+    1. The [individual command files](slash-commands), containing their definitions and functionality.
+    2. The command handler, which dynamically reads the files and executes the commands.
+    3. The [command deployment script](command-deployment), to register your slash commands with Discord so they appear in the interface.
+
+    These steps can be done in any order, but **all are required** before the commands are fully functional.
+
+    This page details how to complete **Step 2**. Make sure to also complete the other pages linked above!
+
+</Alert>
+
+## Loading command files
+
+Now that your command files have been created, your bot needs to load these files on startup.
+
+In your _`index.js`_ file, make these additions to the base template:
+
+<CH.Code rows={14}>
+
+```js JavaScript focus=1:3,9
+import { readdir } from 'node:fs/promises';
+import { join } from 'node:path';
+import { fileURLToPath } from 'node:url';
+// focus[18:28]
+import { Client, Collection, Events, GatewayIntentBits } from 'discord.js';
+import config from './config.json' assert { type: 'json' };
+
+const client = new Client({ intents: [GatewayIntentBits.Guilds] });
+
+const commands = new Collection();
+
+client.once(Events.ClientReady, () => {
+	console.log('Ready!');
+});
+```
+
+```ts TypeScript focus=1:3,9:14
+import { readdir } from 'node:fs/promises';
+import { join } from 'node:path';
+import { fileURLToPath } from 'node:url';
+// focus[18:28]
+import { Client, Collection, Events, GatewayIntentBits } from 'discord.js';
+import config from './config.json';
+
+const client = new Client({ intents: [GatewayIntentBits.Guilds] });
+
+interface CommandModule {
+	data: RESTPostAPIChatInputApplicationCommandsJSONBody;
+	execute(interaction: ChatInputCommandInteraction): Promise<void>;
+}
+
+const commands = new Collection<string, CommandModule>();
+
+client.once(Events.ClientReady, () => {
+	console.log('Ready!');
+});
```

Comment:
This doesn't seem to appear on the preview. Is anyone able to reproduce? If so, could this be fixed in another PR?

Reply:
https://github.com/discordjs/discord.js/pull/9465